src/Servers/Kestrel/Core/src/Internal/Http/ReasonPhrases.cs

```diff
@@ -82,155 +82,95 @@ private static byte[] CreateStatusBytes(int statusCode)
+                StatusCodes.Status403Forbidden => _bytesStatus403,
+                StatusCodes.Status404NotFound => _bytesStatus404,
+                StatusCodes.Status405MethodNotAllowed => _bytesStatus405,
+                StatusCodes.Status406NotAcceptable => _bytesStatus406,
+                StatusCodes.Status407ProxyAuthenticationRequired => _bytesStatus407,
+                StatusCodes.Status408RequestTimeout => _bytesStatus408,
+                StatusCodes.Status409Conflict => _bytesStatus409,
+                StatusCodes.Status410Gone => _bytesStatus410,
+                StatusCodes.Status411LengthRequired => _bytesStatus411,
+                StatusCodes.Status412PreconditionFailed => _bytesStatus412,
+                StatusCodes.Status413PayloadTooLarge => _bytesStatus413,
+                StatusCodes.Status414UriTooLong => _bytesStatus414,
+                StatusCodes.Status415UnsupportedMediaType => _bytesStatus415,
+                StatusCodes.Status416RangeNotSatisfiable => _bytesStatus416,
+                StatusCodes.Status417ExpectationFailed => _bytesStatus417,
+                StatusCodes.Status418ImATeapot => _bytesStatus418,
+                StatusCodes.Status419AuthenticationTimeout => _bytesStatus419,
+                StatusCodes.Status421MisdirectedRequest => _bytesStatus421,
+                StatusCodes.Status422UnprocessableEntity => _bytesStatus422,
+                StatusCodes.Status423Locked => _bytesStatus423,
+                StatusCodes.Status424FailedDependency => _bytesStatus424,
+                StatusCodes.Status426UpgradeRequired => _bytesStatus426,
+                StatusCodes.Status428PreconditionRequired => _bytesStatus428,
+                StatusCodes.Status429TooManyRequests => _bytesStatus429,
+                StatusCodes.Status431RequestHeaderFieldsTooLarge => _bytesStatus431,
+                StatusCodes.Status451UnavailableForLegalReasons => _bytesStatus451,
+
+                StatusCodes.Status500InternalServerError => _bytesStatus500,
+                StatusCodes.Status501NotImplemented => _bytesStatus501,
+                StatusCodes.Status502BadGateway => _bytesStatus502,
+                StatusCodes.Status503ServiceUnavailable => _bytesStatus503,
+                StatusCodes.Status504GatewayTimeout => _bytesStatus504,
+                StatusCodes.Status505HttpVersionNotsupported => _bytesStatus505,
+                StatusCodes.Status506VariantAlsoNegotiates => _bytesStatus506,
+                StatusCodes.Status507InsufficientStorage => _bytesStatus507,
+                StatusCodes.Status508LoopDetected => _bytesStatus508,
+                StatusCodes.Status510NotExtended => _bytesStatus510,
+                StatusCodes.Status511NetworkAuthenticationRequired => _bytesStatus511,
+
+                _ => null
+            };
+
+            if (candidate is not null && (string.IsNullOrEmpty(reasonPhrase) || WebUtilities.ReasonPhrases.GetReasonPhrase(statusCode) == reasonPhrase))
```

Comment:
When can `WebUtilities.ReasonPhrases.GetReasonPhrase(statusCode)` ever be equal to anything other than `candidate` or `string.Empty`? If it is possible, we should Probably use `string.Equals(..., StringComparison.Ordinal)`.

Reply:
> When can WebUtilities.ReasonPhrases.GetReasonPhrase(statusCode) ever be equal to anything other than candidate or string.Empty? 

It can't. I added a test to ensure that any status code defined in `WebUtilities.ReasonPhrases` is also defined in here as a possible candidate.

The fast-path here is `candidate is not null` and `reasonPhrase is null`. We only perform the `Dictionary` lookup if a custom reason phrase is specified.

If this condition fails, it means there is either no candidate OR the custom status code is not the default one, so we create a new one.

I don't see how this could regress the perf for the common case, unless cq is somehow terrible for the different switch statement.

Reply:
Makes sense. Can this be updated to the follwoing?

```suggestion
            if (candidate is not null && (string.IsNullOrEmpty(reasonPhrase) || string.Equals(WebUtilities.ReasonPhrases.GetReasonPhrase(statusCode), reasonPhrase, StringComparison.Ordinal))
```

You'll probably also need to add a `using System;`

Reply:
Done. I assume this is for readability?

Reply:
Yes, but I made this comment before microbenchmarking the two methods. On small equivalent strings like "OK", the difference is pretty substantial. Rather than waste more of your time, I undid this change and force pushed. I'll get this merged. Thanks!

|              Method |      Mean |     Error |    StdDev |
|-------------------- |----------:|----------:|----------:|
| StringEqualsOrdinal | 0.7248 ns | 0.0004 ns | 0.0003 ns |
|  StringDoubleEquals | 0.3780 ns | 0.0139 ns | 0.0130 ns |